service/model.py

```diff
@@ -154,10 +155,16 @@ def find_or_404(cls, by_id):
 
     @classmethod
     def find_by_id_status(cls, by_id: int, by_status=True):
-        """ Find active recommendations of a product [id] """
+        """ Find [status: active/inactive] recommendations of a [product: id] """
         cls.logger.info("Processing lookup for id %s with status %s", by_id, by_status)
         return cls.query.filter(cls.id==by_id, cls.status==by_status)
 
+    @classmethod
+    def find_by_id_type(cls, by_id: int, by_type: int):
+        """ Find recommendations of a [product: id] with [type: typeid] """
```

Comment:
Could you please raise TypeErrors when by_id or by_type are not integers and raise a DataValidationError when by_type is not in [1, 2, 3]

Reply:
Oh okay sure. Thank you for reminding!